<launch>
<!-- from realsense_ros -->
  <arg name="offline" default="false"/>
  <arg name="pos_lim" default="10"/>
  <arg name="z_offset" default="0"/>
  <arg name="max_distance" default="3000"/>
  <arg name="min_distance" default="100"/>

  <include unless="$(arg offline)" 
    file="$(find realsense2_camera)/launch/rs_camera.launch">
    <arg name="align_depth" value="true"/>
    <arg name="linear_accel_cov" value="1.0"/>
    <arg name="unite_imu_method" value="linear_interpolation"/>
  </include>
  
  <node pkg="rgbdp_video" name="depth_converter" type="depth_converter.py" output="screen">
    <param name="pos_lim" value="$(arg pos_lim)"/> 
    <param name="z_offset" value="$(arg z_offset)"/> 
    <param name="max_distance" value="$(arg max_distance)"/> 
    <param name="min_distance" value="$(arg min_distance)"/> 
    <param name="base_frame" value="map"/> 
  </node>

  <node pkg="imu_filter_madgwick" type="imu_filter_node" name="ImuFilter">
    <param name="use_mag" type="bool" value="false" />
    <param name="_publish_tf" type="bool" value="false" />
    <param name="_world_frame" type="string" value="enu" />
    <remap from="/imu/data_raw" to="/camera/imu"/>
  </node>

  <include file="$(find rtabmap_ros)/launch/rtabmap.launch">
    <arg name="args" value="--delete_db_on_start"/>
    <arg name="rgb_topic" value="/camera/color/image_raw"/>
    <arg name="depth_topic" value="/camera/aligned_depth_to_color/image_raw"/>
    <arg name="camera_info_topic" value="/camera/color/camera_info"/>
    <arg name="depth_camera_info_topic" value="/camera/depth/camera_info"/>
    <arg name="rtabmapviz" value="false"/>
    <arg name="rviz" value="true"/>
    <arg name="rviz_cfg" value="$(find rgbdp_video)/launch/config/rgbdp_video.rviz" />
  </include>
  <param name="rtabmap/rgbd_odometry/Odom/ResetCountdown" type="string" value="15"/> 

<!--
  <include file="$(find robot_localization)/launch/ukf_template.launch"/>
  <param name="/ukf_se/frequency" value="300"/>
  <param name="/ukf_se/base_link_frame" value="camera_link"/>
  <param name="/ukf_se/odom0" value="rtabmap/odom"/>
  <rosparam param="/ukf_se/odom0_config">[true,true,true,
                                          true,true,true,
                                          true,true,true,
                                          true,true,true,
                                          true,true,true]
  </rosparam>
  <param name="/ukf_se/odom0_relative" value="true"/>
  <param name="/ukf_se/odom0_pose_rejection_threshold" value="10000000"/>
  <param name="/ukf_se/odom0_twist_rejection_threshold" value="10000000"/>

  <param name="/ukf_se/imu0" value="/imu/data"/>
  <rosparam param="/ukf_se/imu0_config">[false, false, false,
                                         true,  true,  true,
                                         true,  true,  true,
                                         true,  true,  true,
                                         true,  true,  true]
  </rosparam>
  <param name="/ukf_se/imu0_differential" value="true"/>
  <param name="/ukf_se/imu0_relative" value="false"/>
  <param name="/ukf_se/use_control" value="false"/>
-->
  <!-- <param name="/ukf_se/odom0_config" value="{true,true,true,}"/> -->
</launch>
